src/posix/platform/multicast_routing.cpp

```diff
@@ -309,6 +309,8 @@ otError MulticastRoutingManager::AddMulticastForwardingCache(const Ip6::Address
     }
     else
     {
+        VerifyOrExit(otIp6PrefixMatch(&aSrcAddr, otThreadGetMeshLocalEid(gInstance)) < OT_IP6_PREFIX_BITSIZE,
```

Comment:
All prefixes that are not off-mesh routable should be blocked?

Reply:
According to spec 9.4.7.3.1, it only required the source packet not to be Mesh-Local.